keras_core/ops/core_test.py

```diff
@@ -205,6 +205,23 @@ def body(x, y):
         self.assertAllClose(x, np.ones((2, 3)) * 6)
         self.assertAllClose(y, np.ones((3, 2)) * 6)
 
+    def test_fori_loop(self):
+        def body_fun(i, x):
+            return x + i
+
+        initial_value = np.array(0)
+        result = core.fori_loop(0, 10, body_fun, initial_value)
+        self.assertAllClose(result, 45)
+
+    def test_fori_loop_shape_inference(self):
```

Comment:
Move the shape inference to the test class for shape inference (they're all together)